include/solvers/postprocessing_velocities.h

```diff
@@ -0,0 +1,91 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2019 -  by the Lethe authors
+ *
+ * This file is part of the Lethe library
+ *
+ * The Lethe library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+
+ *
+ * Author: Audrey Collard-Daigneault, Polytechnique Montreal, 2020 -
+ */
+
+#ifndef lethe_postprocessing_velocities_h
+#define lethe_postprocessing_velocities_h
+
+
+// Lac - Trilinos includes
+#include <deal.II/lac/trilinos_parallel_block_vector.h>
+#include <deal.II/lac/trilinos_vector.h>
+
+// Dofs
+#include <deal.II/dofs/dof_accessor.h>
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_tools.h>
+
+// Numerics
+#include <deal.II/numerics/solution_transfer.h>
+#include <deal.II/numerics/vector_tools.h>
+
+// Lethe Includes
+#include <core/parameters.h>
+#include <core/simulation_control.h>
+
+#include "navier_stokes_solver_parameters.h"
+#include "post_processors.h"
+
+
+using namespace dealii;
+/**
+ * @brief AverageVelocities. The AverageVelocities class calculates the
+ * time-averaged velocities and pressure (<u>, <v>, <w>, <p>). The generated
+ * vector is output with the solution and visualization is possible.
+ */
+template <int dim, typename VectorType, typename DofsType>
+class AverageVelocities
+{
+public:
+  /**
+   * @brief calculate_average_velocities. This function calculates time-averaged
+   * velocities and pressure with dof vector with no ghost cell.
+   *
+   * @param local_evaluation_point. The vector solutions with no ghost cells
+   *
+   * @param simulation_control. The simulation information (time)
```

Comment:
Why do you pass the entire simulation_control if the only things you need are the time? I am just wondering.

Reply:
It can be changed